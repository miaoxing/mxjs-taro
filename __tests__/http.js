import $ from 'miaoxing';
import Taro from '@tarojs/taro';
import '../';
import {createPromise} from '@mxjs/test';

Taro.getStorageSync = jest.fn();
Taro.showToast = jest.fn().mockImplementation(() => {
  return {};
});

describe('http', () => {
  test('patch', async () => {
    const promise = createPromise();
    Taro.request = jest.fn().mockImplementationOnce(() => promise.resolve({
      statusCode: 200,
    }));

    await $.http({
      method: 'patch',
      url: 'test',
    });

    expect(Taro.request.mock.calls).toMatchSnapshot();
  });
});
