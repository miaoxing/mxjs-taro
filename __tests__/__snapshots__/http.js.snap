// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`http patch 1`] = `
Array [
  Array [
    Object {
      "complete": [Function],
      "header": Object {
        "Accept": "application/json, text/plain, */*",
        "X-HTTP-Method-Override": "PATCH",
      },
      "method": "POST",
      "url": "test",
    },
  ],
]
`;
